### PR TITLE
fix: get machine ID from inside of machineTokenInfo (SC-319)

### PR DESCRIPTION
**Pull request by renanrodrigo**
CS is going to send us the machine ID inside of `machineTokenInfo`, and we should try to get it from there.

## Desired commit type::
 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist:
 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: uaclient/util.py [hunk 1]
```diff
@@ -299,8 +299,10 @@ def get_machine_id(cfg) -> str:
     # Docker images do not define ETC_MACHINE_ID or DBUS_MACHINE_ID on trusty
     # per Issue: #489
 
-    if cfg.machine_token:
-        cfg_machine_id = cfg.machine_token.get("machineId")
+    if isinstance(cfg.machine_token, dict):
```
Comment (lucasmoura) [inline uaclient/util.py]: Why we need the `isinstance` check here ?

Comment (renanrodrigo) [reply uaclient/util.py]: we don't: there was a failing test, but its related to some mock behavior. Already removed.

File: uaclient/contract.py [hunk 2]
```diff
@@ -221,14 +221,14 @@ def _request_machine_token_update(
...
-                response.get("machineId", data.get("machineId", "")),
+            machine_id = response.get("machinetTokenInfo", {}).get(
+                "machineId", data.get("machineId")
             )
+            self.cfg.write_cache("machine-id", machine_id)
         return response
 
     def _get_platform_data(self, machine_id):
-        """"Return a dict of platform-relateddata for contract requests"""
+        """Return a dict of platform-relateddata for contract requests"""
```
Comment (lucasmoura) [inline uaclient/contract.py:231]: I think we can add an extra space here between related and data

File: uaclient/tests/test_util.py [hunk 3]
```diff
@@ -575,9 +573,10 @@ def test_fallback_used_if_all_other_files_are_empty(
         self, FakeConfig, tmpdir, empty_value
     ):
         data_machine_id = tmpdir.join("machine-id")
-        cfg = FakeConfig.for_attached_machine()
-        # Cache it in cfg, so the load_file mock does not mess everything up
-        cfg.machine_token
+        cfg = FakeConfig()
```
Comment (lucasmoura) [inline uaclient/tests/test_util.py:576]: Can't we just do something like this here:
```python
FakeConfig().for_attached_machine(
    machine_token={"test": "blah"}
)
```

File: uaclient/contract.py (hunk 2 above)
Comment (renanrodrigo) [reply uaclient/contract.py:231]: Oh black got this one. Will fix

File: uaclient/tests/test_util.py (hunk 3 above)
Comment (renanrodrigo) [reply uaclient/tests/test_util.py:576]: :man_facepalming:

Comment (renanrodrigo) [reply uaclient/tests/test_util.py]: had to initialize the property as well because of the mock, but it's clearer for sure

**Review by lucasmoura — APPROVED** [review]

**Comment by renanrodrigo** [comment]
LXD integration tests dying randomly and unrelated

**Comment by lucasmoura** [comment]
Couldn't reproduce the failing tests locally

**Pull request merged**